client/client.go

```diff
@@ -163,3 +166,46 @@ func (client Mikrotik) getMikrotikClient() (c *routeros.Client, err error) {
 
 	return
 }
+
+func boolToMikrotikBool(b bool) string {
+	if b {
+		return "yes"
+	} else {
+		return "no"
+	}
+}
+
+func Marshal(s interface{}) string {
+	rv := reflect.ValueOf(s)
+	elem := rv.Elem()
+
+	if rv.Kind() != reflect.Ptr {
+		panic("Command attribute construction cannot work without a pointer")
+	}
+
+	var attributes []string
+
+	for i := 0; i < elem.NumField(); i++ {
+		value := elem.Field(i)
+		fieldType := elem.Type().Field(i)
+		// supports multiple struct tags--assumes first is mikrotik field name
+		tag := strings.Split(fieldType.Tag.Get("mikrotik"), ",")[0]
```

Comment:
This has the potential to panic if a caller passes in a struct without a `mikrotik` tag. I think we should make sure this function does not panic in that case (and returns an empty string).

Reply:
I actually think it's fine because of `tag != ""` in the if statement. I also added a unit test feeding in a struct without `mikrotik` tags, and it doesn't cause a panic.